src/io.c

```diff
@@ -154,13 +155,15 @@ RETSIGTYPE IO_SIGCHLDHandler( int whichsig )
     retcode = waitpid(-1, &status, WNOHANG);
     if (retcode > 0) {   /* One of our child processes terminated */
         if (WIFEXITED(status) || WIFSIGNALED(status)) {
-            if (!statsfull) {
-                stats[lastats] = status;
-                pids[lastats++] = retcode;
-                if (lastats >= maxstats) lastats = 0;
-                if (lastats == fistats) statsfull = 1;
-            } else
-                Pr("#E Overflow in table of terminated processes\n",0,0);
+            if(!CheckChildStatusChanged(retcode, status)) {
+                if (!statsfull) {
+                    stats[lastats] = status;
+                    pids[lastats++] = retcode;
+                    if (lastats >= maxstats) lastats = 0;
+                    if (lastats == fistats) statsfull = 1;
+                } else
+                    Pr("#E Overflow in table of terminated processes\n",0,0);
+            }
```

Comment:
In `FuncIO_WaitPid`, very similar code exists. Should that perhaps also call `CheckChildStatusChanged`?